bee-ternary/src/lib.rs

```diff
@@ -0,0 +1,875 @@
+    ///
+    /// # Panics
+    ///
+    /// This function will panic if the slice is not byte-aligned
+    ///
+    /// # Safety
+    ///
+    /// This function is marked `unsafe` because modification of the trit slice in a manner that is
+    /// not valid for this encoding is undefined behaviour.
+    pub unsafe fn as_i8_slice_mut(&mut self) -> &mut [i8] {
+        self.0.as_i8_slice_mut()
+    }
+
+    /// Fetch the trit at the given index of this trit slice without first checking whether the
+    /// index is in bounds. Providing an index that is not less than the length of this slice is
+    /// undefined behaviour.
+    ///
+    /// This is perhaps the 'least bad' `unsafe` function in this crate: not because any form of
+    /// undefined behaviour is better or worse than another (after all, the point of undefined
+    /// behaviour is that it is undefined) but because it's the easiest to use correctly.
+    ///
+    /// # Safety
+    ///
+    /// An index with a value less then the result of [`Trits::len`] must be used. Any other value
+    /// is undefined behaviour.
+    pub unsafe fn get_unchecked(&self, index: usize) -> T::Trit {
+        debug_assert!(
+            index < self.0.len(),
+            "Attempt to get trit at index {}, but length of slice is {}",
+            index,
+            self.len(),
+        );
+        self.0.get_unchecked(index)
+    }
+
+    /// Set the trit at the given index of this trit slice without first checking whether the
+    /// index is in bounds. Providing an index that is not less than the length of this slice is
+    /// undefined behaviour.
+    ///
+    /// This is perhaps the 'least bad' `unsafe` function in this crate: not because any form of
+    /// undefined behaviour is better or worse than another (after all, the point of undefined
+    /// behaviour is that it is undefined) but because it's the easiest to use correctly.
+    ///
+    /// # Safety
+    ///
+    /// An index with a value less then the result of [`Trits::len`] must be used. Any other value
+    /// is undefined behaviour.
+    pub unsafe fn set_unchecked(&mut self, index: usize, trit: T::Trit) {
+        debug_assert!(
+            index < self.0.len(),
+            "Attempt to set trit at index {}, but length of slice is {}",
+            index,
+            self.len(),
+        );
+        self.0.set_unchecked(index, trit);
+    }
+
+    /// Fetch the trit at the given index of this trit slice, if the index is valid.
+    pub fn get(&self, index: usize) -> Option<T::Trit> {
+        if index < self.0.len() {
+            unsafe { Some(self.get_unchecked(index)) }
+        } else {
+            None
+        }
+    }
+
+    /// Set the trit at the given index of this mutable trit slice, if the index is valid.
+    ///
+    /// # Panics
+    ///
+    /// This function will panic if the index is not less than the length of this slice.
+    // TODO: Should we return `Option<()>` instead?
```

Comment:
Has this been answered ?

Reply:
Not yet. I'm conservatively going with panicking behaviour for now.